tensorflow/contrib/learn/python/learn/ops/autoencoder_ops.py

```diff
@@ -0,0 +1,56 @@
+"""TensorFlow ops for autoencoder."""
+#  Copyright 2015-present The Scikit Flow Authors. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.ops import nn
+from tensorflow.python.ops import rnn_cell
+from tensorflow.python.ops import variable_scope as vs
+from tensorflow.contrib.learn.python.learn.ops import dropout_ops
+
+
+def autoencoder(tensor_in, hidden_units,
```

Comment:
hm, usually auto-encoders are split into encoder and decoder, where decoder is inverse of encoder (for example encoder can be 100, 50, 10 hidden units, decoder - 10, 50, 100). 
It may be make sense to have separate encoder / decoder parts and then have DNNAutoEncoder vs LSTMAutoEncoder vs etc.